data-reporting-service/investigation-service/src/test/java/gov/cdc/etldatapipeline/investigation/InvestigationDataProcessingTests.java

```diff
@@ -0,0 +1,273 @@
+        confirmationMethod.setConfirmationMethodTime("2024-01-15T10:20:57.647");
+
+        transformer.transformInvestigationData(investigation);
+        verify(kafkaTemplate, times(2)).send(topicCaptor.capture(), keyCaptor.capture(), messageCaptor.capture());
+        assertEquals(CONFIRMATION_TOPIC, topicCaptor.getValue());
+
+        Function<InvestigationConfirmationMethod, List<String>> cmDetailsFn = (m) -> Arrays.asList(
+                String.valueOf(m.getPublicHealthCaseUid()),
+                m.getConfirmationMethodCd(),
+                m.getConfirmationMethodDescTxt(),
+                m.getConfirmationMethodTime());
+
+        Function<InvestigationConfirmationMethodKey, List<String>> cmKeyFn = (k) -> Arrays.asList(
+                String.valueOf(k.getPublicHealthCaseUid()),
+                k.getConfirmationMethodCd());
+
+        assertTrue(containsWords.apply(keyCaptor.getValue(), cmKeyFn.apply(confirmationMethodKey)));
+        assertTrue(containsWords.apply(messageCaptor.getValue(), cmDetailsFn.apply(confirmationMethod)));
+    }
+
+    @Test
+    public void testObservationNotificationIds() {
+        Investigation investigation = new Investigation();
+
+        investigation.setPublicHealthCaseUid(investigationUid);
+        investigation.setObservationNotificationIds(readFileData(FILE_PREFIX + "ObservationNotificationIds.json"));
+        transformer.investigationNotificationOutputTopicName = NOTIFICATION_TOPIC;
+        transformer.investigationObservationOutputTopicName = OBSERVATION_TOPIC;
+
+        InvestigationNotification notification = new InvestigationNotification();
+        notification.setPublicHealthCaseUid(investigationUid);
+        notification.setNotificationId(263748597L);
+
+        InvestigationObservation observation = new InvestigationObservation();
+        observation.setPublicHealthCaseUid(investigationUid);
+        observation.setObservationId(263748596L);
+
+        transformer.transformInvestigationData(investigation);
+        verify(kafkaTemplate, times (2)).send(topicCaptor.capture(), keyCaptor.capture(), messageCaptor.capture());
+        assertEquals(OBSERVATION_TOPIC, topicCaptor.getValue());
+
+        Function<InvestigationNotification, List<String>> nDetailsFn = (n) -> Arrays.asList(
+                String.valueOf(n.getPublicHealthCaseUid()),
+                String.valueOf(n.getNotificationId()));
+
+        Function<InvestigationObservation, List<String>> oDetailsFn = (o) -> Arrays.asList(
+                String.valueOf(o.getPublicHealthCaseUid()),
+                String.valueOf(o.getObservationId()));
+
+        String actualCombined = String.join(" ",messageCaptor.getAllValues());
+
+        assertTrue(containsWords.apply(actualCombined, nDetailsFn.apply(notification)));
+        assertTrue(containsWords.apply(actualCombined, oDetailsFn.apply(observation)));
+    }
+
+    @Test
+    public void testInvestigationReporting() {
```

Comment:
If you want you can separate out the tests as two files between model transformations and model deserialization.